tajo-pullserver/src/main/java/org/apache/tajo/pullserver/FadvisedFileRegion.java

```diff
@@ -158,9 +160,9 @@ public void releaseExternalResources() {
    * we don't need the region to be cached anymore.
    */
   public void transferSuccessful() {
-    if (PullServerUtil.isNativeIOPossible() && manageOsCache && getCount() > 0) {
+    if (PullServerUtil.isNativeIOPossible() && manageOsCache && count() > 0) {
```

Comment:
Can you add file checking ? 
if (PullServerUtil.isNativeIOPossible() && manageOsCache && count() > 0 && super.isOpen())

It will fix the "bad file descriptor"
```
2015-03-03 10:34:40,755 WARN org.apache.tajo.pullserver.PullServerUtil: Failed to manage OS cache for /data05/tajo/data/q_1425346386770_0001/output/1/hash-shuffle/3/263
java.lang.NullPointerException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.apache.tajo.pullserver.PullServerUtil.posixFadviseIfPossible(PullServerUtil.java:56)
        at org.apache.tajo.pullserver.FadvisedFileRegion.transferSuccessful(FadvisedFileRegion.java:163)
        at org.apache.tajo.pullserver.FileCloseListener.operationComplete(FileCloseListener.java:46)
```